src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxQueryEvaluator.cs

```diff
@@ -0,0 +1,297 @@
+                
+                case (OperatorType.GreaterThanEqual, ValueTokenType.Double):
+                    return _searcher.GreaterThanOrEqual(previousMatch, fieldId, (double)fieldValue);
+                case (OperatorType.GreaterThanEqual, ValueTokenType.Long):
+                    return _searcher.GreaterThanOrEqual(previousMatch, fieldId, (long)fieldValue);
+                case (OperatorType.GreaterThanEqual, ValueTokenType.String):
+                    Slice.From(_allocator, fieldValue.ToString(), out sliceValue);
+                    return _searcher.GreaterThanOrEqual(previousMatch, fieldId, sliceValue);
+                
+                default:
+                    throw new Exception($"Got {type} and the value: {field.ValueType} at UnaryMatch.");
+            }
+        }
+
+        private IQueryMatch EvaluateBetween(BetweenExpression betweenExpression, bool negated)
+        {
+            var exprMin = GetValue(betweenExpression.Min);
+            var exprMax = GetValue(betweenExpression.Max);
+            var fieldId = GetFieldIdInIndex(GetFieldName((FieldExpression)betweenExpression.Source));
+            
+            switch (exprMin.ValueType, exprMax.ValueType, negated)
+            {
+                case (ValueTokenType.Long, ValueTokenType.Long, false):
+                    return _searcher.Between(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                case (ValueTokenType.Long, ValueTokenType.Long, true):
+                    return _searcher.NotBetween(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                
+                case (ValueTokenType.Double, ValueTokenType.Double, false):
+                    return _searcher.Between(_searcher.AllEntries(), fieldId, (double)exprMin.FieldValue, (double)exprMax.FieldValue);
+                case (ValueTokenType.Double, ValueTokenType.Double, true):
+                    return _searcher.NotBetween(_searcher.AllEntries(), fieldId, (double)exprMin.FieldValue, (double)exprMax.FieldValue);
+                
+                case (ValueTokenType.String, ValueTokenType.String, false):
+                    return _searcher.Between(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                case (ValueTokenType.String, ValueTokenType.String, true):
+                    return _searcher.NotBetween(_searcher.AllEntries(), fieldId, (long)exprMin.FieldValue, (long)exprMax.FieldValue);
+                
+                
+                default:
+                    var unsupportedType = exprMin.ValueType is ValueTokenType.Double or ValueTokenType.Long or ValueTokenType.String
+                        ? exprMax.ValueType
+                        : exprMin.ValueType;
+                    throw new InvalidDataException($"Got {unsupportedType} but expected: {ValueTokenType.String}, {ValueTokenType.Long}, {ValueTokenType.Double}.");
```

Comment:
Also need to mention that the types here must match, can't do an int and string, frex

Reply:
Such error is handled at the parsing stage.
`ThrowIncompatibleTypesOfVariables`